### DISCUSSION TITLE
Is the lack of admin rights an issue?

### DISCUSSION THREAD

**Discussion by samharrison7** [ukceh-rse/community#13] [Science Computing Group issues]
I'm interested to know how people feel about a lack of admin rights, now that we've had some time to settle in to not having them.

A few thoughts from my perspective:
* Other institutes grant people admin rights securely by given them VLANs. UKCEH has done this in the past. Why isn't this a solution anymore? VLANs would be compatible with CyberEssentials - there are even [blog posts highlighting this](https://www.indelibledata.co.uk/cyber-essentials/how-vlans-can-help-you-achieve-cyber-essentials-compliance/).
* No admin rights makes testing new software (that requires admin rights) really cumbersome. We can't just `sudo apt install` something to quickly test it out, we have to get someone else to do it for us.
* The solution of using container software like Podman works for some cases, but has its issues:
    * There is a fairly steep learning curve, particularly if you're never used containers before (which I'm guessing is the majority of our scientific computing user base). 
    * There are container-in-a-container issues when you might need to use some software that uses containers inside the container. This is possible, but not without its difficulties, which adds extra overhead to development time.
* All of this pushes people to using their own personal devices instead of UKCEH devices.

**Comment by mattjbr123** [comment]
Thanks Sam - this captures my understanding as well. 
Fits with other feedback I've received that too many arguably simple things are requiring IT tickets to solve

I know @metazool has asked about a linux-based solution to this, forgive me I can't remember the details or the name right now, which IIRC received positive noises

**Comment by jmarshrossney** [comment]
I'm finding this really difficult, and would go so far as to say I feel the part of my job which is supporting others with their research code is largely compromised.

The reason it's such a problem for me is that the significant task of understanding how to set up something that could reasonably be called a development environment is not something I've been able to allocate time for since joining CEH. There is quite fairly an expectation to 'get shit done' and this inevitably leads me to work on my linux desktop at home.

(For a long time I could create neither `conda` nor `venv` environments on my work computer so needed a VM to do any Python work, no matter how small. They seem to have changed something recently to allow `venv`.)

So since I do not use CEH infrastructure as a daily driver I'm guessing just as much as anyone else when someone asks me for help with datalabs, VMs etc.

All that said I take responsibility for not having reached out to the relevant people for a 'train the trainers' session. That's probably a good place to start, but I do think the restrictiveness is a fundamental problem.

**Reply by samharrison7** [reply]
Thanks Joe, really useful thoughts.

This raises another point about different experiences for different people. For example, I've got a functioning `conda` setup on my laptop and that enables me to get around a lot of admin issues. But I know plenty of people who haven't been able to get this far (like yourself), or have really had to battle to do so. I had to have custom Carbon Black rules rules to get it working for me, which periodically need adapting, and all of this takes *time*.
